src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -0,0 +1,55 @@
+from .simstat import SimStat
+from .statistic import Scalar, Distribution, Accumulator
+from .group import Group, Vector
+import json
+
+def json_loader(path: str) -> SimStat:
+    
+    with open(path) as f:
+        simstat_obj = json.load(f, object_hook=__json_to_simstat)
+
+    return simstat_obj
+
+def __json_to_simstat(d: dict) -> SimStat:
+    print(d)
+    if 'type' in d:
+        if d['type'] == 'Scalar':
+            return Scalar(
+                          value=d.get('value'),
```

Comment:
Why `d.get` and not `d['value']`?

Reply:
Some of these statistics have optional data members, and .get() supports returning NoneType (rather than throwing an error). I made all of them .get() for the sake of continuity, but I could differentiate between optional (d.get()) and non-optional data members (d['value']) if you like

Reply:
I see! Would it work to just do the following?

```python
return Scalar(**d)
```

Reply:
I don't think so. These statistics don't take **kwargs as an argument.

Reply:
Let me try it and see what happens though.

Reply:
All python functions take a dictionary as a parameter :smile:

See, for instance, https://book.pythontips.com/en/latest/args_and_kwargs.html

Reply:
I believe you... it works but it requires removing the "type" from the dictionary d, just because of the way we've defined the constructor for statistics and Vectors. I'll push an updated version for you soon to check out

Reply:
Makes sense. One thing we could to is add `**kwargs` as the final parameter to the stats objects to allow other kwargs.